polygeist/lib/Dialect/Polygeist/Analysis/MemoryAccessAnalysis.cpp

```diff
@@ -1075,7 +1060,37 @@ OffsetVector::getConstIntegerValue(size_t row, DataFlowSolver &solver) const {
 // MemoryAccess
 //===----------------------------------------------------------------------===//
 
+MemoryAccessMatrix
+MemoryAccess::getIntraThreadAccessMatrix(unsigned numThreads) const {
+  assert(numThreads < matrix.getNumColumns() &&
```

Comment:
Not currently because we collect memory accesses that are in a  loop nest so there should be at least one loop IV. Even if the memory access doesn't use the loop IV its memory access matrix will have numColumns = num thread + num loop IVs. 

I can change it though to <=

Reply:
Changed